src/gui/ogr/qgsvectorlayersaveasdialog.cpp

```diff
@@ -58,6 +59,11 @@ QgsVectorLayerSaveAsDialog::QgsVectorLayerSaveAsDialog( QgsVectorLayer *layer, O
   }
   setup();
 
+  if ( layer && leLayername->isEnabled() )
+  {
+    leLayername->setText( QgsMapLayerUtils::launderLayerName( layer->name() ) );
```

Comment:
```suggestion
    leLayername->setPlaceholderText( QgsMapLayerUtils::launderLayerName( layer->name() ) );
```

The empty handling is already implemented and this will allow a user to revert to the default handling in case of accidental keystrokes or similar

Reply:
Done -- I've gone with QgsFilterLineEdit with the default layer name now available if you reset the layer name text box :+1: